manifests/resource/vhost.pp

```diff
@@ -1,9 +1,24 @@
 define nginx::resource::vhost (
-  String                      $source,
+  String                      $source  = '',
+  String                      $content = '',
   Enum[ 'present', 'absent' ] $ensure  = present,
   Boolean                     $enabled = true,
 ) {
 
+  if (empty($source)) and (empty($content)) {
```

Comment:
This check should verify that only one of source or content is specified as they should be mutually exclusive.